01-js-week-1/ruby-to-js-worksheet.md

```diff
@@ -0,0 +1,865 @@
+# Ruby to JavaScript Worksheet
+Read the code in each section, then write the equivalent JavaScript code for the Ruby that is given. Start by writing it out in a text editor or on a piece of paper. Then try running it and tweak your code until it successfully runs with expected output.
+
+Each problem stands alone. Variables from previous problems do not exist.
+
+## Problem Set
+
+### !challenge
+
+* type: code-snippet
+* language: javascript
+* id: 10c8b04b-6428-42d9-88cc-f405c8206306
+* title: Problem 1
+<!-- * points: [1] (optional, the number of points for scoring as a checkpoint) -->
+<!-- * topics: [python, pandas] (optional the topics for analyzing points) -->
+
+##### !question
+
+Convert the following ruby code into JavaScript using the input box below.
+
+```ruby
+ada_age = 2
+
+if person_age < ada_age
+    print "This person is younger"
+elsif ada_age < person_age
+    print "Ada is younger"
+else
+    print "They’re the same!"
+end
+```
+
+##### !end-question
+
+##### !placeholder
+
+```js
+function ageChecker(personAge) {// DO NOT EDIT THIS CODE
+  //Paste your code here
+
+
+}//DO NOT EDIT THIS ONE EITHER
+```
+
+##### !end-placeholder
+
+##### !tests
+
+```js
+/**
+ * Chris doesn't like this testing format and I don't know it well enough to ignore him!
+ * 
+ * @param impl - What the real function is supposed to do.
+ */
+function fn (impl = () => { }) {
+  const mockFn = function (...args) {
+    mockFn.mock.calls.push(args);
+    mockFn.mock.instances.push(this);
+    try {
+      const value = impl.apply(this, args); // call impl, passing the right this
+      mockFn.mock.results.push({ type: 'return', value });
+      return value; // return the value
+    } catch (value) {
+      mockFn.mock.results.push({ type: 'throw', value });
+      throw value; // re-throw the error
+    }
+  }
+  mockFn.mock = { calls: [], instances: [], results: [] };
+  return mockFn;
+}
+
+describe('ageChecker', () => {
+  let oldConsoleLog = console.log;
+  
+  // Mock console.log
+  beforeEach(() => {
+    console.log = fn();
+  });
+  
+  // restore console.log
+  afterEach(() => {
+    console.log = oldConsoleLog;
+  });
+
+  it("Prints correctly for older person (person_age = 35)", () => {
+    ageChecker(35);
+    expect(console.log.mock.calls.length).to.equal(1);
+    expect(console.log.mock.calls[0][0]).to.equal("Ada is younger");
+  });
+  it("Prints correctly for younger person (person_age = 0)", () => {
+    ageChecker(0);
+    expect(console.log.mock.calls.length).to.equal(1);
+    expect(console.log.mock.calls[0][0]).to.equal("This person is younger");
+  });
+  it("Prints correctly for same age person (person_age = 2)", () => {
+    ageChecker(2);
+    expect(console.log.mock.calls.length).to.equal(1);
+    expect(console.log.mock.calls[0][0]).to.equal("They’re the same!");
```

Comment:
Just because of the difference between '   and ’  

```suggestion
    expect(console.log.mock.calls[0][0]).to.match(/They[\’\']re the same!/);
```